lib/ansible/modules/cloud/vmware/vmware_guest_video.py

```diff
@@ -0,0 +1,325 @@
+            if self.params['memory_3D_mb'] is not None:
+                memory_3d = self.params['memory_3D_mb']
+                if not self.is_power_of_2(memory_3d):
+                    self.module.fail_json(msg="memory_3D_mb attribute should be an integer value and power of 2(32-2048).")
+                else:
+                    if memory_3d < 32 or memory_3d > 2048:
+                        self.module.fail_json(msg="memory_3D_mb attribute should be an integer value and power of 2(32-2048).")
+                if memory_3d * 1024 != video_card_facts['memory_3D']:
+                    video_spec.device.graphicsMemorySizeInKB = memory_3d * 1024
+                    self.change_detected = True
+        else:
+            if self.params['renderer_3D'] is not None or self.params['memory_3D_mb'] is not None:
+                self.module.fail_json(msg='3D renderer or 3D memory can not be configured if 3D is not enabled.')
+        if not self.change_detected:
+            return None
+        return video_spec
+
+    def reconfigure_vm_video(self, vm_obj):
+        """
+        Reconfigure video card settings of virtual machine
+        Args:
+            vm_obj: Managed object of virtual machine
+        Returns: Reconfigure results
+        """
+        video_card_spec = self.get_video_card_spec(vm_obj)
+        if video_card_spec is None:
+            return {'changed': False, 'failed': False, 'instance': self.video_card_facts}
+        self.config_spec.deviceChange.append(video_card_spec)
+        try:
+            task = vm_obj.ReconfigVM_Task(spec=self.config_spec)
+            wait_for_task(task)
+        except vim.fault.InvalidDeviceSpec as invalid_device_spec:
+            self.module.fail_json(msg="Failed to configure video card on given virtual machine due to invalid"
+                                      " device spec : %s" % (to_native(invalid_device_spec.msg)),
+                                  details="Please check ESXi server logs for more details.")
+        except vim.fault.RestrictedVersion as e:
+            self.module.fail_json(msg="Failed to reconfigure virtual machine due to"
+                                      " product versioning restrictions: %s" % to_native(e.msg))
+        if task.info.state == 'error':
+            return {'changed': self.change_detected, 'failed': True, 'msg': task.info.error.msg}
+        video_card_facts = self.gather_video_card_facts(vm_obj)[1]
+        return {'changed': self.change_detected, 'failed': False, 'instance': video_card_facts}
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        name=dict(type='str'),
+        uuid=dict(type='str'),
+        folder=dict(type='str'),
+        datacenter=dict(type='str', default='ha-datacenter'),
+        gather_video_facts=dict(type='bool', default=False),
+        use_auto_detect=dict(type='bool'),
+        display_number=dict(type='int'),
+        video_memory_mb=dict(type='int'),
```

Comment:
If the minimum value is `1.17` as per module docs, shouldn't this be a float?
```suggestion
        video_memory_mb=dict(type='float'),
```